eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/client/EclairRpcClient.scala

```diff
@@ -385,15 +434,21 @@ class EclairRpcClient(val instance: EclairInstance)(
   }
 
   override def getReceivedInfo(
-      paymentHash: Sha256Digest): Future[ReceivedPaymentResult] = {
-    eclairCall[ReceivedPaymentResult]("getreceivedinfo",
-                                      "paymentHash" -> paymentHash.hex)
-  }
-
-  override def getReceivedInfo(
-      invoice: LnInvoice): Future[ReceivedPaymentResult] = {
-    eclairCall[ReceivedPaymentResult]("getreceivedinfo",
-                                      "invoice" -> invoice.toString)
+      paymentHash: Sha256Digest): Future[Option[ReceivedPaymentResult]] = {
```

Comment:
nit: I think it would be cleaner to have `ReceivedPaymentResult` be a `sealed trait` which was extended by the two possible cases

Reply:
i agree, but with it being time pressing i think we should just use `Option` for this for now.